ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/tabsontop/v1/tabsontop/clientlibs/site/js/tabs.js

```diff
@@ -148,32 +132,31 @@
             return this.element.querySelector(Tabs.selectors.qm);
         }
 
+        #getTabListElement() {
+            return this.element.querySelector(Tabs.selectors.olTabList)
+        }
+
         #syncTabLabels(){
-            var tabs = this._elements["tab"];
-            if (tabs) {
-                if(Array.isArray(tabs)){
-                    for (var i = 0; i < tabs.length; i++) {
-                        tabs[i].id = this._elements["tabPanelElementId"][i] + "__tab";
-                        tabs[i].setAttribute("aria-controls", this._elements["tabPanelElementId"][i] + "__tabpanel");
-                    }
-                }else{
-                    tabs.id = this._elements["tabPanelElementId"][0] + "__tab";
-                    tabs.setAttribute("aria-controls", this._elements["tabPanelElementId"][0] + "__tabpanel");
+            var tabs = this.#getCachedTabs();
+            var tabPanels = this.#getCachedTabPanels();
+            if (tabPanels) {
+                for (var i = 0; i < tabPanels.length; i++) {
+                    var childViewId = tabPanels[i].querySelectorAll(
```

Comment:
this can result in NPE.
pls check for null before resolving the id.

Reply:
@Nitin, currently tab panel can't exist without a child element, if that would be the case then sightly itself will break.

Reply:
one may overlay the sightly but not js. lets not couple them if not called out explicitly in contract.

Reply:
Hi @nit23uec , not yet sure this is the way to go, lots of things will break if ids are removed from tabpanels corresponding to child elements. Repeatability is completely dependent upon it, also the removal of duplication of ids is also dependent on it.

Its a small check but don't you think this approach might not be correct. We can't possibly handle all the scenarios where sightly is overlayed and doesn't conform to OOTB sightly but still the  OOTB view is used as is.

If you still think that we should add a check, i will add it.

cc:- @sakshi-arora1